kratos/includes/node.h

```diff
@@ -757,42 +757,60 @@ class Node : public Point,  public IndexedObject, public Flags
         typename DofsContainerType::iterator it_begin = mDofs.begin();
         typename DofsContainerType::iterator it_end = mDofs.end();
         typename DofsContainerType::iterator it;
-        //if the guess is exact return the guess
-        if(pos < it_end-it_begin)
-        {
+        // If the guess is exact return the guess
+        if(pos < it_end-it_begin) {
             it = it_begin + pos;
-            if( (it)->GetVariable() == rDofVariable)
-            {
+            if( (it)->GetVariable() == rDofVariable) {
                 return *it;
             }
         }
 
         // Otherwise do a find
-        it = mDofs.find(rDofVariable.Key());
-        if ( it!= mDofs.end() )
-        {
-            return *it;
+        return GetDof(rDofVariable);

```

Comment:
here you must access again to begin and end, so it is slower... i would leave the implementaion you had before...

Reply:
I'm not sure if it is slower... std vector has begin and end pointers as member so optimizer would substitute it. Nevertheless you may not trust to optimizer which is understandable....